### ISSUE TITLE
Daemon crash in logging, node does not shut down properly

### ISSUE THREAD

**Issue by gz-c** [skycoin/skycoin#838]
There is a crash in the logging library.  This may be due to the additional log buffering added.

Secondly, when this crash occurs in the daemon loop, the client does not shut down properly. Only the daemon shuts down. The error must not be handled correctly from `cmd/skycoin/skycoin.go`

Version 0.21.1

```
[skycoin.daemon:ERROR] recover:runtime error: slice bounds out of range                                                                                          │····························
 stack:goroutine 23 [running]:                                                                                                                                   │····························
runtime/debug.Stack(0xc421bbb1f8, 0x8a2640, 0xc43510)                                                                                                            │····························
        /home/skyd/go/src/runtime/debug/stack.go:24 +0x79                                                                                                        │····························
github.com/skycoin/skycoin/src/daemon.(*Daemon).Run.func1()                                                                                                      │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/src/daemon/daemon.go:320 +0x9a                                                                          │····························panic(0x8a2640, 0xc43510)                                                                                                                                        │····························
        /home/skyd/go/src/runtime/panic.go:489 +0x2cf                                                                                                            │····························bytes.(*Buffer).Write(0xc420300408, 0xc4206c9c00, 0x38, 0x400, 0x38, 0x0, 0x0)                                                                                   │····························
        /home/skyd/go/src/bytes/buffer.go:138 +0xc9                                                                                                              │····························io.(*multiWriter).Write(0xc42042d340, 0xc4206c9c00, 0x38, 0x400, 0xc52bc0, 0x0, 0x0)                                                                             │····························
        /home/skyd/go/src/io/multi.go:60 +0x72                                                                                                                   │····························log.(*Logger).Output(0xc420258320, 0x5, 0xc42153cfc0, 0x37, 0x0, 0x0)                                                                                            │····························
        /home/skyd/go/src/log/log.go:168 +0x210                                                                                                                  │····························github.com/skycoin/skycoin/vendor/github.com/op/go-logging.(*LogBackend).Log(0xc42042e7e0, 0x5, 0x3, 0xc421927560, 0xc701a0, 0xc421927500)                       │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/vendor/github.com/op/go-logging/log_nix.go:72 +0x27e                                                    │····························github.com/skycoin/skycoin/vendor/github.com/op/go-logging.(*moduleLeveled).Log(0xc4202cafc0, 0x5, 0x2, 0xc421927560, 0xc4212d3340, 0x0)                         │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/vendor/github.com/op/go-logging/level.go:116 +0xf2                                                      │····························github.com/skycoin/skycoin/vendor/github.com/op/go-logging.(*Logger).log(0xc42016c3f0, 0x5, 0x9341d3, 0x16, 0xc4212d3340, 0x2, 0x2)                              │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/vendor/github.com/op/go-logging/logger.go:162 +0x1cb                                                    │····························github.com/skycoin/skycoin/vendor/github.com/op/go-logging.(*Logger).Debug(0xc42016c3f0, 0x9341d3, 0x16, 0xc4212d3340, 0x2, 0x2)                                 │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/vendor/github.com/op/go-logging/logger.go:239 +0x66                                                     │····························
github.com/skycoin/skycoin/src/daemon.(*GetBlocksMessage).Process(0xc4212d3020, 0xc420300300)                                                                    │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/src/daemon/visor.go:612 +0x299                                                                          │····························github.com/skycoin/skycoin/src/daemon.(*Daemon).processMessageEvent(0xc420300300, 0xb5b560, 0xc4212d3020, 0xc4212d30a0)                                          │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/src/daemon/daemon.go:743 +0x86                                                                          │····························
github.com/skycoin/skycoin/src/daemon.(*Daemon).Run(0xc420300300, 0x0, 0x0)                                                                                      │····························
        /home/skyd/gopath/src/github.com/skycoin/skycoin/src/daemon/daemon.go:494 +0x16f3                                                                        │····························
main.Run.func5(0xc4201c23b0, 0xc420300300)                                                                                                                       │····························
        /home/skyd/skycoin/cmd/skycoin/skycoin.go:630 +0x51                                                                                                      │····························
created by main.Run                                                                                                                                              │····························
        /home/skyd/skycoin/cmd/skycoin/skycoin.go:631 +0x8c2                                                                                                     │····························
                                                                                                                                                                 │····························
[skycoin.daemon:INFO] Daemon closed 
```

**Comment by gz-c** [comment]
The daemon crash error handling was fixed in #791 
https://github.com/skycoin/skycoin/commit/0cef92e3cee6daf012531d1127e43146e47aaf7c#diff-c68ecdf08891bcc11a71ec8c1bdfd99dR676

**Comment by gz-c** [comment]
Seems to be fixed with log buffer removal
